scripts/md2html/build.sh

```diff
@@ -2,46 +2,80 @@
 
 # Author: @MikeRalphson
 
-# run this script from the root of the repo. It is designed to be run by a GitHub workflow.
-# It contains bashisms
+# run this script from the root of the repo
+# It is designed to be run by a GitHub workflow
 
-mkdir -p deploy/oas
-mkdir -p deploy/js
+# Usage: build.sh [version | "latest" | "src"]
+# When run with no arguments, it builds artifacts for all published specification versions.
+# It may also be run with a specific version argument, such as "3.1.1" or "latest"
+# Finally, it may be run with "src" to build "src/oas.md"
+#
+# It contains bashisms
 
-cd scripts/md2html
+if [ -z "$1" ]; then
+  deploydir="deploy/oas"
+else
+  deploydir="deploy-preview"
+fi
 
-cp -p ../../node_modules/respec/builds/respec-w3c.* ../../deploy/js/
+mkdir -p $deploydir/js
+mkdir -p $deploydir/temp
 
-latest=`git describe --abbrev=0 --tags`
-latestCopied=none
+latest=$(git describe --abbrev=0 --tags)
+latestCopied="none"
 lastMinor="-"
-for filename in $(ls -1 ../../versions/[23456789].*.md | sort -r) ; do
-  if [[ ${filename} == *-editors.md ]];then
-    continue
-  fi
 
-  version=$(basename "$filename" .md)
+if [ -z "$1" ]; then
+  specifications=$(ls -1 versions/[23456789].*.md | grep -v -e "\-editors" | sort -r)
+elif [ "$1" = "latest" ]; then
+  specifications=$(ls -1 versions/$latest.md)
+elif [ "$1" = "src" ]; then
+  specifications="src/oas.md"
+else
+  specifications=$(ls -1 versions/$1.md)
+fi
+
+cp -p node_modules/respec/builds/respec-w3c.* $deploydir/js/
+
+for specification in $specifications; do
+  version=$(basename $specification .md)
   minorVersion=${version:0:3}
-  tempfile=../../deploy/oas/v$version-tmp.html
-  echo -e "\n=== v$version ==="
+  tempfile="$deploydir/temp/$version.html"
 
-  node md2html.js --maintainers ../../versions/$version-editors.md ${filename} > $tempfile
-  npx respec --use-local --src $tempfile --out ../../deploy/oas/v$version.html
+  if [ "$1" = "src" ]; then
+    destination="$deploydir/$version.html"
+    maintainers="EDITORS.md"
+  else
+    destination="$deploydir/v$version.html"
+    maintainers="$(dirname $specification)/$version-editors.md"
+  fi
```

Comment:
Move this up and set `deploydir` to `deploy-preview` only for "src".

Reply:
I moved this as far up the block as I could, but as these values depend on `$version`, they need to be defined in-loop.